apps/anoma_node/lib/node/transaction/backends.ex

```diff
@@ -381,4 +392,89 @@ defmodule Anoma.Node.Transaction.Backends do
 
     EventBroker.event(event)
   end
+
+  @spec cairo_resource_tx(String.t(), binary(), Noun.t()) ::
+          :ok | :error
+  defp cairo_resource_tx(node_id, id, result) do
+    with {:ok, tx} <- CTransaction.from_noun(result),
+         true <- Anoma.RM.Transaction.verify(tx),
+         true <- root_existence_check(tx, node_id, id),
+         # No need to check the commitment existence
+         true <- nullifier_existence_check(tx, node_id, id) do
+      {ct, append_roots} =
+        case Ordering.read(node_id, {id, :ct}) do
```

Comment:
Are we using the same commitment tree/nullifier and commitment set as the transparent RM?

Reply:
Assuming they go to different commitment trees and nullifier sets for now, things may change after interoperability discussion.
Would using the `id` here result in different storage? Or should I use a separate indicator like `:cairo_nullifiers` to distinguish the nullifier sets?

Reply:
So yes, you would have to use different keys for things of different sorts. If you want to keep transparent and shielded resources separate, use `:your_name_for_commitment_tree` as the key at which you read and write/append.

`id` here is just the id of the transaction, it does not result in different storage, every transaction fed into a node X will use one storage that node X provides, yet different resource machines can use different keys.

Reply:
Actually, here is a good idea: write a test where the same node processes a cairo and transparent transaction. Do all the checks pass?